web/modules/custom/farm_nfa/src/PlanPermissionsRouteChecker.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Drupal\farm_nfa;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\user\Entity\Role;
+
+/**
+ * Checks access for displaying plan permissions route.
+ */
+class PlanPermissionsRouteChecker implements AccessInterface {
+
+  /**
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   Run access checks for this account.
+   */
+  public function access(AccountInterface $account, $plan) {
+    // Load the entity by ID.
+    $entity = \Drupal::entityTypeManager()->getStorage('plan')->load($plan);
+    $users = $entity->get('user')->referencedEntities();
```

Comment:
It would be safer if you check if the field exists with hasField and if it is not empty with isEmpty